### ISSUE TITLE
Status doesn't properly load extended tweets from streaming APIs

### ISSUE THREAD

**Issue by robmathers** [bear/python-twitter#506]
## Summary
The `Status.NewFromJsonDict` method doesn't properly load extended tweets with an `extended_tweet` payload, [which is how the streaming APIs provide extended tweets](https://dev.twitter.com/overview/api/upcoming-changes-to-tweets)

## Steps to Reproduce
1. Load a JSON dict of an extended tweet in Twitter's "Compatibility with additional `extended_tweet` in payload" mode. You can find a [sample here](https://github.com/twitterdev/tweet-updates/blob/master/samples/initial/compatibilityplus_extended_13997.json), or use `GetStreamSample()` and check for tweets with `truncated = True`.
2. Use `Status.NewFromJsonDict` to convert the JSON dict to a `Status`.
3. The status will just have a `text` property, without a `full_text`, even when the API instance has `tweet_mode` set to `extended`. Other extended entities are also not available.

## Workaround
By copying the keys from the JSON's `extended_tweet` object to the root object, `NewFromJsonDict` properly parses the tweets as an extended tweet, with `full_text` available.

### Example:
```
# Where json_dict is a dictionary parsed from the JSON of a tweet in this mode
if json_dict.has_key('extended_tweet'):
    for key, value in json_dict['extended_tweet'].items():
        json_dict[key] = value
converted_tweet = Status.NewFromJsonDict(json_dict)
# converted_tweet.full_text will be populated and not truncated

**Comment by robmathers** [comment]
This may be related to issue #491 as well.

**Comment by zefoo** [comment]
I'm not able to get this to work. Any updates on this? I have a search result and I try something like this:

result.NewFromJsonDict(result.AsDict()).full_text

And it seems to be empty. Any ideas?

Thank you!

**Comment by jeremylow** [comment]
Fix lands in 564e6bb620b20d09d937917e3094bcdb2c82c812
